modules/end-user-portal/src/Pages/Apply/Organisation.cshtml.cs

```diff
@@ -0,0 +1,82 @@
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Core.Config;
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Core.Data.Interfaces;
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Core.HttpClientServices.Interfaces;
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Models;
+using GpConnect.NationalDataSharingPortal.EndUserPortal.Models.Response;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+
+namespace GpConnect.NationalDataSharingPortal.EndUserPortal.Pages.Apply;
+
+public partial class OrganisationModel : BaseModel
+{
+    private readonly ITempDataProviderService _tempDataProviderService;
+    private readonly IOrganisationLookupService _organisationLookupService;
+
+    public OrganisationModel(IOptions<ApplicationParameters> applicationParameters, IOrganisationLookupService organisationLookupService, ITempDataProviderService tempDataProviderService) : base(applicationParameters)
+    {
+        _tempDataProviderService = tempDataProviderService;
+        _organisationLookupService = organisationLookupService;
+    }
+
+    public IActionResult OnGetAsync()
+    {
+        ClearModelState();
+        PrepopulateOrganisationDetails();
+        return Page();
+    }
+
+    private void PrepopulateOrganisationDetails()
+    {
+        if (_tempDataProviderService.GetItem<OrganisationResult>("Organisation") != null)
+        {
+            OrganisationResult = _tempDataProviderService.GetItem<OrganisationResult>("Organisation");
+            SiteOdsCode = OrganisationResult.OdsCode;
+            OrganisationFound = true;
+        }
+    }
+
+    public async Task<IActionResult> OnPostFindOrganisationAsync()
+    {
+        if (!ModelState.IsValid)
+        {
+            return Page();
+        }
+        if (!_tempDataProviderService.HasItems)
+        {
+            return RedirectToPage("./Timeout");
+        }
+        var organisationResult = await GetOrganisationDetails(SiteOdsCode);
+        if (organisationResult != null)
+        {
+            OrganisationFound = true;
+            OrganisationResult = organisationResult;
+            _tempDataProviderService.PutItem("Organisation", organisationResult);
+        }
+        return Page();
+    }
+
+    private async Task<OrganisationResult?> GetOrganisationDetails(string siteOdsCode)
+    {
+        var organisationResult = await _organisationLookupService.GetOrganisationAsync(siteOdsCode);
+        return organisationResult;
```

Comment:
return without assignment?

Reply:
I don't understand what you mean?

Reply:
This method could be something like...
`return _organisationLookupService.GetOrganisationAsync(siteOdsCode)`

as we are not using the returned object from GetOrganisationAsync, there is no value to creating the callback handler. so may as well return the Task